### PR TITLE
Enable Server Communications

### PR DESCRIPTION
**Pull request by paullin03**
* Add pushes from FirebaseCloudMessaging
* Add Retrofit for Server Login and Session APIs

### CODE REVIEW

File: frontend/app/src/main/java/com/quickpick/MainActivity.java [hunk 1]
```diff
@@ -31,44 +45,48 @@ private void registerButtons() {
...
         });
+    }
+
+    private void showAlertDialog() {
+        AlertDialog dialog = new AlertDialog.Builder(MainActivity.this)
+                .setTitle(getString(R.string.enter_session_id_title))
+                .setView(R.layout.dialog_join_session)
+                .create();
+        dialog.show();
+        EditText sessionCode = dialog.findViewById(R.id.session_code_edit_text);
```
Comment (richardechegaray) [inline frontend/app/src/main/java/com/quickpick/MainActivity.java:75]: maybe some whitespace here? kind of chunky

File: frontend/app/src/main/java/com/quickpick/repositories/SessionRepository.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+            }
+
+            @Override
+            @EverythingIsNonNull
+            public void onFailure(Call<SessionPayload> call, Throwable t) {
+                Log.d(SESSION, call.request().toString(), t);
+            }
+        });
+    }
+}
```
Comment (richardechegaray) [inline frontend/app/src/main/java/com/quickpick/repositories/SessionRepository.java:63]: very clean i love it

File: frontend/app/src/main/res/layout/activity_session.xml [hunk 3]
```diff
@@ -6,4 +6,12 @@
     android:layout_height="match_parent"
     tools:context=".SessionActivity">
 
+    <TextView
+        android:id="@+id/session_key_text_view"
```
Comment (richardechegaray) [inline frontend/app/src/main/res/layout/activity_session.xml:10]: we should be consistent w our namings - decide if we wanna do _text or _text_view or _textview - i think i did _text on mine

File: frontend/app/src/main/res/layout/dialog_join_session.xml [hunk 4]
```diff
@@ -0,0 +1,39 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:gravity="center_horizontal"
+    android:orientation="vertical">
+
+    <EditText
+        android:id="@+id/session_code_edit_text"
```
Comment (richardechegaray) [inline frontend/app/src/main/res/layout/dialog_join_session.xml:9]: but i see you used _text for a non textview so maybe ill change mine then

**Review by richardechegaray — COMMENTED** [review]
initial thoughts submitted!

File: frontend/app/src/main/res/layout/dialog_join_session.xml (hunk 4 above)
Comment (paullin03) [reply frontend/app/src/main/res/layout/dialog_join_session.xml:9]: Let's stick with text for regular TextView and edit_text for EditText

File: frontend/app/src/main/java/com/quickpick/MainActivity.java (hunk 1 above)
Comment (paullin03) [reply frontend/app/src/main/java/com/quickpick/MainActivity.java:77]: Added some whitespace

**Review by richardechegaray — APPROVED** [review]

**Pull request merged**